pilot/pkg/serviceregistry/kube/queue.go

```diff
@@ -70,64 +60,55 @@ func NewQueue(errorDelay time.Duration) Queue {
 		delay:   errorDelay,
 		queue:   make([]Task, 0),
 		closing: false,
-		lock:    sync.Mutex{},
+		cond:    sync.NewCond(&sync.Mutex{}),
 	}
 }
 
 func (q *queueImpl) Push(item Task) {
-	q.lock.Lock()
+	q.cond.L.Lock()
+	defer q.cond.L.Unlock()
 	if !q.closing {
 		q.queue = append(q.queue, item)
 	}
-	q.lock.Unlock()
+	q.cond.Signal()
 }
 
 func (q *queueImpl) Run(stop <-chan struct{}) {
 	go func() {
 		<-stop
-		q.lock.Lock()
+		q.cond.L.Lock()
 		q.closing = true
-		q.lock.Unlock()
+		q.cond.L.Unlock()
 	}()
 
-	rate := os.Getenv(enableQueueThrottleEnv)
 	rateLimit := 100
-	if len(rate) > 0 {
-		r, err := strconv.Atoi(rate)
-		if err == nil {
-			rateLimit = r
-		}
-	}
-	// Throttle processing up to smoothed 10 qps with bursts up to 100 qps
-	var rateLimiter flowcontrol.RateLimiter
-	if rateLimit > 0 {
-		rateLimiter = flowcontrol.NewTokenBucketRateLimiter(float32(rateLimit), 10*rateLimit)
-	}
+	rateLimiter := flowcontrol.NewTokenBucketRateLimiter(float32(rateLimit), 10*rateLimit)
```

Comment:
Well, we need to remove the reatelimiter - not the option to configure it to be faster...

Reply:
Can we add back the option, and set it to 0 by default ( i.e. disable the extra rate limit )

If we have the Cond we don't need the ratelimit.